### PR TITLE
Ticket #7854 - detect media type using mimemagic gem

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ before_install:
 - docker run -d -p 9000:9000 -e "MINIO_ACCESS_KEY=AKIAIOSFODNN7EXAMPLE" -e "MINIO_SECRET_KEY=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" minio/minio server /data
 - sudo apt-get -qq update
 - sudo apt-get install wget openjdk-7-jre redis-server imagemagick fontconfig libfontconfig
-- sudo apt-get install ffmpeg gstreamer0.10-ffmpeg ffmpegthumbnailer -y
+- sudo apt-get install ffmpegthumbnailer -y
```
Comment (infojunkie) [inline .travis.yml]: we don't need `ffmpeg` anymore ?

File: app/models/bot/smooch.rb [hunk 2]
```diff
@@ -712,6 +713,15 @@ def self.create_project_media(message, type, extra)
     pm
   end
 
+  def self.detect_media_type(message)
```
Comment (infojunkie) [inline app/models/bot/smooch.rb]: I prefer the style of coding where we avoid side-effects on the input parameter, by explicitly returning the value - in this case the MIME type. E.g. `message['mediaType'] = self.detect_media_type(message)`

File: test/models/bot/smooch_test.rb [hunk 3]
```diff
@@ -69,9 +69,11 @@ def setup
     @media_url = 'https://smooch.com/image/test.jpeg'
     @media_url_2 = 'https://smooch.com/image/test2.jpeg'
     @video_url = 'https://smooch.com/video/test.mp4'
+    @video_ur_2 = 'https://smooch.com/video/test2.mp4'
```
Comment (infojunkie) [inline test/models/bot/smooch_test.rb]: is this a file that's not really a video?

File: app/models/bot/smooch.rb [hunk 4]
```diff
@@ -600,6 +600,7 @@ def self.save_message(message_json, app_id)
          when 'video'
            self.save_media_message(message, 'video')
          when 'file'
+           self.detect_media_type(message)
            m = message['mediaType'].to_s.match(/^(image|video)\//)
```
Comment (infojunkie) [inline app/models/bot/smooch.rb]: question to @melsawy and @caiosba : for security reasons, should we reject, or at least log, files whose MIME type differs between our detection and Smooch? Or are we not worried about their contents?

Comment (caiosba) [reply app/models/bot/smooch.rb]: I think we should log (to Errbit) but still save it

Comment (infojunkie) [reply app/models/bot/smooch.rb]: I agree to log, but I'd rather use the regular log and use a WARNING level. Maybe we need to agree on log vs. Errbit.

File: .travis.yml (hunk 1 above)
Comment (melsawy) [reply .travis.yml]: Yes Karim just need ffmpegthumbnailer only

File: app/models/bot/smooch.rb (hunk 4 above)
Comment (melsawy) [reply app/models/bot/smooch.rb]: For now I'll use a regular log with WARNING is this OK?

Comment (infojunkie) [reply app/models/bot/smooch.rb]: sounds good

File: .travis.yml (hunk 1 above)
Comment (infojunkie) [reply .travis.yml]: oh because `ffmpegthumbnailer` already depends on `ffmpeg` :+1:

File: app/models/bot/smooch.rb (hunk 2 above)
Comment (melsawy) [reply app/models/bot/smooch.rb]: Done

File: test/models/bot/smooch_test.rb (hunk 3 above)
Comment (melsawy) [reply test/models/bot/smooch_test.rb]: Done

**Pull request merged**